### PR TITLE
Set instance variables at initialization to avoid spamming rake warnings.

### PR DESCRIPTION
**Pull request by qingling128**
Fixes https://github.com/googleapis/google-auth-library-ruby/issues/227.

### CODE REVIEW

File: lib/googleauth/signet.rb [hunk 1]
```diff
@@ -38,6 +38,11 @@ module OAuth2
     # This reopens Client to add #apply and #apply! methods which update a
     # hash with the fetched authentication token.
     class Client
+      def initialize
+        super
+        @refresh_listeners = nil
```
Comment (dazuma) [inline lib/googleauth/signet.rb]: This one is not going to work. We're re-opening an existing class `Signet::OAuth2::Client` from the `signet` gem. This `initialize` method will _replace_ the method from the original class, which we don't want to do. (And the `super` calls the initializer on the superclass which happens to be `Object`.) Instead, I think we're going to need to find the uses of `@refresh_listeners` and guard them with, for example, `if defined? @refresh_listeners`.

**Comment by qingling128** [comment]
@dazuma - Just pushed another commit to address the feedback. Let me know if you'd prefer me to squash the 2 commits.

File: lib/googleauth/signet.rb (hunk 1 above)
Comment (qingling128) [reply lib/googleauth/signet.rb]: Addressed in https://github.com/googleapis/google-auth-library-ruby/pull/230/commits/f908519d5523db021e5bd17094b5b14ec9b0e2bc.

**Comment by qingling128** [comment]
It said this PR required sync with master, so I just sync'ed it. How can I kick off the tests again? By somehow applying the `kokoro:force-run` label?

File: lib/googleauth/signet.rb [hunk 2]
```diff
@@ -84,7 +84,11 @@ def fetch_access_token! options = {}
       end
 
       def notify_refresh_listeners
-        listeners = @refresh_listeners || []
+        if defined? @refresh_listeners
+          listeners = @refresh_listeners
+        else
+          listeners = []
+        end
```
Comment (dazuma) [inline lib/googleauth/signet.rb]: Looks like Rubocop is flagging this. Prefer:
```ruby
listeners =
  if defined? @refresh_listeners
    @refresh_listeners
  else
    []
  end
```
Or alternately:
```ruby
listeners = defined?(@refresh_listeners) ? @refresh_listeners : []
```

Comment (qingling128) [reply lib/googleauth/signet.rb]: Done. Looks like rubocop is happy now:

```
diff --git a/lib/googleauth/signet.rb b/lib/googleauth/signet.rb
index 0b9f93d..db0dfa8 100644
--- a/lib/googleauth/signet.rb
+++ b/lib/googleauth/signet.rb
@@ -84,11 +84,7 @@ module Signet
       end

       def notify_refresh_listeners
-        if defined? @refresh_listeners
-          listeners = @refresh_listeners
-        else
-          listeners = []
-        end
+        listeners = defined?(@refresh_listeners) ? @refresh_listeners : []
         listeners.each do |block|
           block.call self
         end

google-auth-library-ruby$ rubocop
Inspecting 21 files
.....................

21 files inspected, no offenses detected
```

**Comment by qingling128** [comment]
@dazuma - I just fixed the Rubocop error. Could you help add the `kokoro:force-run` label again to kick off the test?

**Pull request merged**